docs/website/docs/dlt-ecosystem/destinations/qdrant.md

```diff
@@ -0,0 +1,222 @@
+---
+title: Qdrant
+description: Qdrant is a high-performance vector search engine/database that can be used as a destination in dlt.
+keywords: [qdrant, vector database, destination, dlt]
+---
+
+# Qdrant
+
+[Qdrant](https://qdrant.tech/) is an open-source, high-performance vector search engine/database. It deploys as an API service, providing a search for the nearest high-dimensional vectors.
+This destination helps you load data into Qdrant from [dlt resources](../../general-usage/resource.md).
+
+## Setup Guide
+
+1. To use Qdrant as a destination, make sure `dlt` is installed with the `qdrant` extra:
+
+```bash
+pip install dlt[qdrant]
+```
+
+2. Next, configure the destination in the dlt secrets file. The file is located at `~/.dlt/secrets.toml` by default. Add the following section to the secrets file:
+
+```toml
+[destination.qdrant.credentials]
+location = "https://your-qdrant-url"
+api_key = "your-qdrant-api-key"
+```
+
+In this setup guide, we are using the [Qdrant Cloud](https://cloud.qdrant.io/) to get a hosted Qdrant instance and the [FastEmbed](https://github.com/qdrant/fastembed) package that is built into the [Qdrant client library](https://github.com/qdrant/qdrant-client) for generating embeddings.
+
+If no configuration options are provided, the default fallback will be `http://localhost:6333` with no API key.
+
+3. Define the source of the data. For starters, let's load some data from a simple data structure:
+
+```python
+import dlt
+from dlt.destinations.qdrant import qdrant_adapter
+
+movies = [
+    {
+        "title": "Blade Runner",
+        "year": 1982,
+    },
+    {
+        "title": "Ghost in the Shell",
+        "year": 1995,
+    },
+    {
+        "title": "The Matrix",
+        "year": 1999,
+    }
+]
+```
+
+4. Define the pipeline:
+
+```python
+pipeline = dlt.pipeline(
+    pipeline_name="movies",
+    destination="qdrant",
+    dataset_name="MoviesDataset",
+)
+```
+
+5. Run the pipeline:
+
+```python
+info = pipeline.run(
+    qdrant_adapter(
+        movies,
+        embed="title",
+    )
+)
+```
+
+6. Check the results:
+
+```python
+print(info)
+```
+
+The data is now loaded into Qdrant.
```

Comment:
Another question regarding this example. After loading the data, how to query it with `client.query`? As in:

```python
# pip install qdrant-client[fastembed]

from qdrant_client import QdrantClient

client = QdrantClient()

result = client.query(
    collection_name="content",
    query_text="Neo",
)
print(result)
```

When I run the above I get:
```
Traceback (most recent call last):
  File "qdrant_query.py", line 23, in <module>
    search_result = client.query(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/qdrant_fastembed.py", line 333, in query
    self.search(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/qdrant_client.py", line 315, in search
    return self._client.search(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/qdrant_remote.py", line 454, in search
    search_result = self.http.points_api.search_points(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/http/api/points_api.py", line 1197, in search_points
    return self._build_for_search_points(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/http/api/points_api.py", line 536, in _build_for_search_points
    return self.api_client.request(
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/http/api_client.py", line 74, in request
    return self.send(request, type_)
  File "/Users/b/dlthub/dlt/.venv/lib/python3.8/site-packages/qdrant_client/http/api_client.py", line 97, in send
    raise UnexpectedResponse.for_response(response)
qdrant_client.http.exceptions.UnexpectedResponse: Unexpected Response: 400 (Bad Request)
Raw response content:
b'{"status":{"error":"Wrong input: Vector params for fast-bge-small-en are not specified in config"},"time":0.002777542}'
```

At the same time `client.search()` works as expected:

```python
query_vector = [.0] * 384

result = client.search(
    collection_name="content",
    query_vector=query_vector,
    limit=3,
)
print(result)

>>> [ScoredPoint(id='00b14076-69bf-536b-aa97-2a98435e4fcc', version=2, score=0.0, payload={'_dlt_id': 'CbMonMJoPDlfKg', '_dlt_load_id': '1699207571.018365', 'title': 'The Matrix', 'year': 1999}, vector=None), ScoredPoint(id='f15418fe-ea92-5b83-98b0-4829cd79b322', version=2, score=0.0, payload={'_dlt_id': 'wgwJEYdunRhldg', '_dlt_load_id': '1699207571.018365', 'title': 'Ghost in the Shell', 'year': 1995}, vector=None), ScoredPoint(id='f3fdeb51-103f-54e1-9fb8-19b4d2abfd8c', version=2, score=0.0, payload={'_dlt_id': 'v4uJ+VeY3gJ83g', '_dlt_load_id': '1699207571.018365', 'title': 'Blade Runner', 'year': 1982}, vector=None)]
```

Reply:
I'll have to update the insertion for this.
The wquery()` method expects a named vector. Where the name is the the name of the model.

Reply:
https://github.com/dlt-hub/dlt/pull/724/commits/86f5b979fbbf0966bbc7820c5f70c10b48b3721a.
You can now perform `query()`. 
And `search()` by specifying the vector name using `<QdrantClient>.get_vector_field_name()`.